app/models/category.rb

```diff
@@ -0,0 +1,5 @@
+class Category < ApplicationRecord
+  belongs_to :parent, class_name: "Category"
+  has_many :children, class_name: "Category", foreign_key: "parent_id"
```

Comment:
`Category"` -> `Category.name`
` "parent_id"` -> :parent_id

`:children`-> xem lại association in rails chú

Reply:
> `:children`-> xem lại association in rails chú
chỗ này có phải e thiếu optional: true k c? @@

Reply:
`has_many` -> thì tên model là số nhìu cậu (haiz) ` has_many :childrens`